commit 72b9956766b26164628ccc979952fd1ae5bc5e41
Author: Kai Uwe Broulik <kde@privat.broulik.de>
Date:   Tue Jun 4 17:47:17 2013 +0200

    Also set minimum size when it *is* constrained
    
    BUG: 309447
    FIXED-IN: 4.11.0

* 2013-06-04: Backport to 4.10.4

diff --git a/plasma/generic/applets/batterymonitor/contents/ui/CompactRepresentation.qml b/plasma/generic/applets/batterymonitor/contents/ui/CompactRepresentation.qml
index f8691a1..e9dc772 100644
--- a/plasma/generic/applets/batterymonitor/contents/ui/batterymonitor.qml	2013-05-28 20:38:22.000000000 +0200
+++ b/plasma/generic/applets/batterymonitor/contents/ui/batterymonitor.qml	2013-06-04 21:15:26.234102844 +0200
@@ -70,8 +70,12 @@
 
             Component.onCompleted: {
                 if (!isConstrained()) {
-                    minimumWidth = 32;
-                    minimumHeight = 32;
+                    minimumWidth = theme.iconSizes.dialog;
+                    minimumHeight = minimumHeight * view.count;
+                } else {
+                    // NOTE: Keep in sync with systray
+                    minimumHeight = 24;
+                    minimumWidth = 24;
                 }
                 plasmoid.addEventListener('ConfigChanged', configChanged);
             }
